src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -95,5 +96,53 @@ public void MulticastOption_CreateSocketSetGetOption_GroupAndInterfaceIndex_SetS
                 Assert.Throws<SocketException>(() => socket.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.AddMembership));
             }
         }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexAnySucceeds()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 0;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index));
+            }
+        }
+
+        [Theory]
+        [InlineData(0)] // Any
+        [InlineData(1)] // Loopback
+        public void MulticastInterface_Set_ValidIndex_Succeeds(int interfaceIndex)
+        {
+            IPAddress multicastAddress = IPAddress.Parse("239.1.2.3");
+            int port = 54321;
```

Comment:
Can this instead bind to any available port?  It's getting "address in use" errors on OS X, and I'm assuming this is the cause.  Same for the other tests.